### PR TITLE
docs: update feature documentations

### LINKED ISSUE #416: Update docs for new/updated features
I completely forgot to mention this during the review of PR #333 and #387, but we should add the inherent documentation to readthedocs about the new IRC and secondary structure features, in particular in [featured.md](https://github.com/DeepRank/deeprank-core/blob/main/docs/features.md) file. 

I think we should also update a bit the description for `electrostatic` and `vanderwaals`, since now we do use a distance cutoff.

### PR DESCRIPTION
**Pull request by DaniBodor**
See [here](https://deeprankcore.readthedocs.io/en/416_features_docs_update_dbodor/features.html) what the site will look like if implementing these changes.

I don't understand why the coveralls is failing. @gcroci2 , any idea?

closes #416

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -2,61 +2,108 @@
...
+
+#### Atom properties:
+These features are only used in atomic graphs.
+
+- `atom_type`: One hot encoding of the atomic element. Options are: C, O, N, S, P, H.
+- `atom_charge`: Atomic charge in Coulomb (float). Taken from `deeprankcore.domain.forcefield.patch.top`.
+- `pdb_occupancy`: Proportion of structures where the atom was detected at this position (float). In some cases a single atom was detected at different positions, in which case separate structures exist whose occupancies sum to 1. Only the highest occupancy atom is used by deeprankcore. 
+
+#### Residue properties:
+- `res_type`: One hot encoding of the amino acid residue (size 20).
```
Comment (gcroci2) [inline docs/features.md:40]:
```suggestion
- `res_type`: One-hot encoding of the amino acid residue (size 20).
```

File: docs/features.md [hunk 2]
```diff
@@ -2,61 +2,108 @@
...
+- `polarity`: One hot encoding of the polarity of the amino acid (options: NONPOLAR, POLAR, NEGATIVE, POSITIVE). Note that sources vary on the polarity for few of the amino acids; see detailed information in `deeprankcore.domain.aminoacidlist.py`.
```
Comment (gcroci2) [inline docs/features.md:41]:
```suggestion
- `polarity`: One-hot encoding of the polarity of the amino acid (options: NONPOLAR, POLAR, NEGATIVE, POSITIVE). Note that sources vary on the polarity for few of the amino acids; see detailed information in `deeprankcore.domain.aminoacidlist.py`.
```

File: docs/features.md [hunk 3]
```diff
@@ -2,61 +2,108 @@
...
+- `res_charge`: The charge of the residue (in fully protonated state) in Coulomb (int). Charge is calculated from summing all atoms in the residue, which results in a charge of 0 for all polar and nonpolar residues, +1 for positive residues and -1 for negative residues.
+- `res_pI`: The isolectric point, i.e. the pH at which the molecule has no net electric charge (float).
+
+- `hb_donors`, `hb_acceptors`: The number of hydrogen bond donor/acceptor atoms in the residue (int). Hydrogen bonds are noncovalent intermolecular interactions formed between an hydrogen atom (partially positively charged) bound to a small, highly electronegative atom (O, N, F) with an unshared electron pair.
+
+#### Properties related to variant residues:
+These features are only used in SingleResidueVariant queries.
+
+- `variant_res`: One hot encoding of variant amino acid (size 20).
+- `diff_charge`, `diff_polarity`, `diff_size`, `diff_mass`, `diff_pI`, `diff_hb_donors`, `diff_hb_acceptors`: Subtraction of the wildtype value of indicated feature from the variant value. For example, if the variant has 4 hb_donors and the wildtype has 5, then `diff_hb_donors == -1`.
```
Comment (gcroci2) [inline docs/features.md:53]:
```suggestion
- `diff_charge`, `diff_polarity`, `diff_size`, `diff_mass`, `diff_pI`, `diff_hb_donors`, `diff_hb_acceptors`: Subtraction of the wildtype value of indicated feature from the variant value (int). For example, if the variant has 4 hb_donors and the wildtype has 5, then `diff_hb_donors == -1`.
```

File: docs/features.md [hunk 4]
```diff
@@ -2,61 +2,108 @@
...
+For residue graphs, the pairwise sum of potentials for all atoms from each residue is used. Note that no distance cutoff is used and the radius of influence is assumed to be infinite, although the potentials tends to 0 at large distance. Also edges are only assigned within a given cutoff radius when graphs are created.
+Nonbond energies are set to 0 for any atom pairs (on the same chain) that are within a cutoff radius of 3.6 Å, as these are assumed to be covalent neighbors or linked by no more than 2 covalent bonds (i.e. 1-3 pairs).
+
+- `electrostatic`: Electrostatic potential (also known as Coulomb potential) between two nodes, calculated using interatomic distances and charges of each atom (float).
+- `vanderwaals`: Van der Waals potential (also known as Lennard-Jones potential) between two nodes, calculated using interatomic distance/s and a list of atoms with vanderwaals parameters (`deeprankcore.domain.forcefield.protein-allhdg5-4_new`). Atom pairs within a cutoff radius of 4.2 Å (but above 3.6 Å) are assumed to be separated by separated by exactly 2 covalent bonds (i.e. 1-4 pairs) and use a set of lower energy parameters.
```
Comment (gcroci2) [inline docs/features.md:109]:
```suggestion
- `vanderwaals`: Van der Waals potential (also known as Lennard-Jones potential) between two nodes, calculated using interatomic distance/s and a list of atoms with vanderwaals parameters (`deeprankcore.domain.forcefield.protein-allhdg5-4_new`, float). Atom pairs within a cutoff radius of 4.2 Å (but above 3.6 Å) are assumed to be separated by separated by exactly 2 covalent bonds (i.e. 1-4 pairs) and use a set of lower energy parameters.
```

File: docs/features.md [hunk 5]
```diff
@@ -2,61 +2,108 @@
...
+## Default node features 
+For atomic graphs, when features relate to residues then _all_ atoms of one residue receive the feature value for that residue.
+
+### Core properties of atoms and residues: `deeprankcore.features.components`
+These features relate to the chemical components (atoms and amino acid residues) of which the graph is composed. Detailed information and descrepancies between sources are described can be found in `deeprankcore.domain.aminoacidlist.py`.
+
+#### Atom properties:
+These features are only used in atomic graphs.
+
+- `atom_type`: One hot encoding of the atomic element. Options are: C, O, N, S, P, H.
```
Comment (gcroci2) [inline docs/features.md:35]:
```suggestion
- `atom_type`: One-hot encoding of the atomic element. Options are: C, O, N, S, P, H.
```

File: docs/features.md [hunk 6]
```diff
@@ -2,61 +2,108 @@
...
+- `res_mass`: The (average) residue mass in Da (float).
+- `res_charge`: The charge of the residue (in fully protonated state) in Coulomb (int). Charge is calculated from summing all atoms in the residue, which results in a charge of 0 for all polar and nonpolar residues, +1 for positive residues and -1 for negative residues.
+- `res_pI`: The isolectric point, i.e. the pH at which the molecule has no net electric charge (float).
+
+- `hb_donors`, `hb_acceptors`: The number of hydrogen bond donor/acceptor atoms in the residue (int). Hydrogen bonds are noncovalent intermolecular interactions formed between an hydrogen atom (partially positively charged) bound to a small, highly electronegative atom (O, N, F) with an unshared electron pair.
+
+#### Properties related to variant residues:
+These features are only used in SingleResidueVariant queries.
+
+- `variant_res`: One hot encoding of variant amino acid (size 20).
```
Comment (gcroci2) [inline docs/features.md:52]:
```suggestion
- `variant_res`: One-hot encoding of variant amino acid (size 20).
```

File: docs/features.md [hunk 7]
```diff
@@ -2,61 +2,108 @@
...
+
+#### Surface accessibility: `deeprankcore.features.surfacearea`
+These features relate to the surface area of the residue, and are computed using [freesasa](https://freesasa.github.io). Note that these features can only be calculated per residue and not per atom.
+
+- `sasa`: [Solvent-Accessible Surface Area](https://en.wikipedia.org/wiki/Accessible_surface_area) is the surface area (in Å^2) of a biomolecule that is accessible to the solvent (float).
+- `bsa`: Buried Surface Area is the surface area (in Å^2) that is buried away from the solvent when two or more proteins or subunits associate to form a complex, i.e. it measures the size of the complex interface (float).
+
+#### Secondary structure: `deeprankcore.features.secondary_structure`
+
+- `sec_struct`: One hot encoding of the [DSSP](https://en.wikipedia.org/wiki/DSSP_(algorithm)) assigned secondary structure of the amino acid, using the three major classes (HELIX, STRAND, COIL). Calculated using [DSSP4](https://github.com/PDB-REDO/dssp).
```
Comment (gcroci2) [inline docs/features.md:79]:
```suggestion
- `sec_struct`: One-hot encoding of the [DSSP](https://en.wikipedia.org/wiki/DSSP_(algorithm)) assigned secondary structure of the amino acid, using the three major classes (HELIX, STRAND, COIL). Calculated using [DSSP4](https://github.com/PDB-REDO/dssp).
```

**Review by gcroci2 — APPROVED** [review]
Very nice additions :)

File: docs/features.md (hunk 3 above)
Comment (DaniBodor) [reply docs/features.md:53]: these are not all ints (e.g. charge is a float, polarity is a list of -1/0/1). They are all of the same type as the original. I would leave the explanation of that out, though, or do you think it's needed?

**Comment by DaniBodor** [comment]
> > See [here](https://deeprankcore.readthedocs.io/en/416_features_docs_update_dbodor/features.html) what the site will look like if implementing these changes.
> > I don't understand why the coveralls is failing. @gcroci2 , any idea?
> > closes #416
> 
> Sometimes it happens, I just rerun the failed job and now it works :)

ah, nice. It actually failed a few times in a row, so thought it was something real, but glad that it's  not.

**Review by gcroci2 — APPROVED** [review]

File: docs/features.md (hunk 3 above)
Comment (gcroci2) [reply docs/features.md:53]: Oh right sorry, no I don't think it's needed to change anything. It's pretty obvious that the type will be the same as the original feature.

**Pull request merged**